coderd/prometheusmetrics/prometheusmetrics.go

```diff
@@ -283,8 +283,177 @@ func Agents(ctx context.Context, logger slog.Logger, registerer prometheus.Regis
 			agentsConnectionLatenciesGauge.Commit()
 			agentsAppsGauge.Commit()
 
+		done:
 			logger.Debug(ctx, "Agent metrics collection is done")
 			metricsCollectorAgents.Observe(timer.ObserveDuration().Seconds())
+
+			ticker.Reset(duration)
+		}
+	}()
+	return cancelFunc, nil
```

Comment:
I believe this function is considered "close" elsewhere, but doesn't give a guarantee that the service has closed. This could e.g. leave running go routines in tests which would be caught by goleak.

(Same with the function below)

It might never be an actual problem but more of a semantic/foolproofing thing. It's never a guarantee though that a function taking a context will exit immediate on cancellation.

Reply:
Would you like me to rename the calling place: `closeAgentStatsFunc` -> `cancelAgentStatsFunc`?

BTW it affects also other functions: `closeUsersFunc`, `closeWorkspacesFunc`, `closeAgentsFunc`

Reply:
Ok, I applied the pattern to all metrics-related functions.